lib/Mojo/mysql/Results.pm

```diff
@@ -9,7 +9,7 @@ has [qw(db sth)];
 
 sub array { ($_[0]->_expand($_[0]->sth->fetchrow_arrayref))[0] }
 
-sub arrays { _c($_[0]->_expand(@{$_[0]->sth->fetchall_arrayref})) }
+sub arrays { _c($_[0]->_types->_expand(@{$_[0]->sth->fetchall_arrayref})) }
```

Comment:
I don't think we should figure out the types, unless we're going to actually expand.

Reply:
I think we have to, because after `fetchall` the statement handle is closed and we can no longer read from it.

Reply:
Then I suggest a bigger change:

```
# The method to call on $sth and true/false for deref
sub array { ($_[0]->_expand(fetchrow_arrayref => 0)[0] }
sub arrays { _c($_[0]->_expand(fetchall_arrayref => 1) }
```

I guess by doing that, we could merge _types() and _expand() again, and maybe even use that true/false flag instead of ref() inside _expand()?